### ISSUE TITLE
Update a downloaded issue

### ISSUE THREAD

**Issue by benoitchantre** [Simbul/baker#638]
I haven't yet tried the newsstand mode, but I suppose this feature is not implemented because there's nothing related to "update" in the Localizable.string

Would it be possible to add that?

A publisher can make some changes and the end user has to be notified if there's an update for a downloaded issue.

An information like "update available" displayed under the title of the issue should be sufficient.

**Comment by folletto** [comment]
Is it even possible in Newsstand? Could you check on Apple's Documentation? :)

**Comment by benoitchantre** [comment]
Yes, I read the Newsstand FAQ and it appears you can update an issue, but only once in a 24h period. The download runs in the background if the user hasn't disabled this feature.
Here's the technical note from Apple: http://developer.apple.com/library/ios/ipad/#technotes/tn2280/_index.html#//apple_ref/doc/uid/DTS40011215

**Comment by nin9creative** [comment]
I don't know if this is possible.  Unless the subscriber deleted the local download and redownloaded it?

**Comment by folletto** [comment]
If you are referring to the question _"The content of my publication needs to be updated throughout the day. Can I initiate a Newsstand background download more than once in a 24-hour period?"_ mind that Apple uses the word "publication" for the whole application and "issue" for the single download. The only thing that FAQ says is that you can't push more than one new issue ("updated publication"), even by saying "If you want your Newsstand app to deliver breaking news, consider sending an issue via background download".

So, up to this point, I still don't think it's possible to update an already released issue.

**Comment by benoitchantre** [comment]
Ok, then it's like a printed publication: once printed and taken from the newsstand, the end user can't have an update if the publisher has changed something. If it's not possible because Apple conceived it like that, we can close this issue and maybe write a small note about that in the wiki. Thanks again for your support.

**Comment by folletto** [comment]
Thanks to you for raising this. :)
And if you or anyone discovers any news on this, please re-open the issue. :D

**Comment by benoitchantre** [comment]
It should be possible, Adobe did it. You can see it for example in Adobe Inspire Magazine.
When you open their app, the shelf refresh the infos about issues (text, images). A notification is displayed if an update is available. Same process when you come back to the shelf from an issue.

I think the update feature they have implemented doesn't rely on the Newsstand library provided by Apple. It's a feature in their app that works the same way if you publish your issues in the newsstand or not.

![image](https://f.cloud.github.com/assets/1521015/87889/c1583ae2-64d5-11e2-9f36-43b2020e4a24.jpg)

**Comment by benoitchantre** [comment]
@folletto can you please reopen this issue?
I think I have no permission to do that?
Thanks

**Comment by folletto** [comment]
Reopened, however I think that unless Baker becomes SaaS or adds a server part that isn't possible.

**Comment by benoitchantre** [comment]
Here's a suggestion that's probably too simple to be realistic
- a part of this feature is to detect an update of an issue: why not store the date of a downloaded issue (from the shelf json) and to compare it again with the date on the server
- if you have something different, that means there's an update
- when an updated issue is detected that way, the shelf displays an update button that removes the issue and downloads the same issue immediately

**Comment by benoitchantre** [comment]
@nin9creative, is it a feature planned in MagRocket?

**Comment by nin9creative** [comment]
I could definitely support such a feature with MagRocket, but it would need Baker front end work to somehow check the server to see if any issues have been updated.  However I would imagine that Adobe isn't constantly checking for updated issues, it's probably relying on some push notification telling their apps that there is an update available for an issue and prompting / reacting accordingly.  That could be accomplished as well.  MagRocket (when Push Notifications are built in) could allow an admin to change the uploaded issue asset and then send out a notification (the nature of the Push Notifications would need to change to support more than just the new newsstand content type) with the issue name or whatever and Baker could react to that to prompt the user to download again.  But then there are exceptions like, the user is subscribed but hasn't yet downloaded - do you want to display the message to them, etc. etc...

I'd need to think about it more, but as @folletto said - you would need a server component that could support the feature to at least provide a mechanism for the Baker shelf to know something was updated.

A simpler approach would be to fully implement alert push notifications in Baker...  Then if you make an update to an issue, you could send that type of notification.  The end user would get the message telling them that a new version of XYZ issue was available and that they should delete/redownload it.  Granted this isn't as slick, but it could work as well as an easier implementation.

**Comment by folletto** [comment]
I should check, however, are we sure you can buy _individual_ magazines outside iOS when you use Adobe? I'm sure you can do that with subscriptions - and that should be already possible with Baker - but individual issues? Is there anyone that can confirm or deny? :)

**Comment by nin9creative** [comment]
I've only read related to subscriptions....  But I am not 100% sure.

**Comment by benoitchantre** [comment]
@nin9creative, the app from Adobe does a refresh of it's data when you open it or when you go back to the library.
You automatically receive a push notification when a new issue is available, but not when there's an update of a downloaded issue.

**Comment by nin9creative** [comment]
Ah gotcha....

**Comment by folletto** [comment]
However, right now:
1. You can just update the magazine, and people can simply archive and re-download it.
2. You can update the JSON file and get the update in the shelf (as of #547).

What's really missing is just the notification for the release of an updated version.

**Comment by nin9creative** [comment]
Exactly

**Comment by folletto** [comment]
What if we check `shelf.json` and if there's a date difference we prompt a new download? Would that work?

**Comment by nin9creative** [comment]
That would work fine I would think...  Simple.

**Comment by benoitchantre** [comment]
Good news :-)
